hack/vf-netns-switcher.sh

```diff
@@ -0,0 +1,343 @@
+    done
+}
+
+switch_interface_vf_representors(){
+    local pf_name="$1"
+    local worker_netns=$2
+
+    if [[ -z "${pf_switch_ids[$pf_name]}" ]] || [[ -z ${pf_port_names[$pf_name]:1} ]];then
+        echo "$pf_name does not have pf_switch_id or pf_port_name, assuming not switchdev..."
+        return 0
+    fi
+
+    for interface in $(ls /sys/class/net);do
+        phys_switch_id=$(cat /sys/class/net/$interface/phys_switch_id)
+        if [[ "$phys_switch_id" != "${pf_switch_ids[$pf_name]}" ]]; then
+            continue
+        fi
+        phys_port_name=$(cat /sys/class/net/$interface/phys_port_name)
+        phys_port_name_pf_index=${phys_port_name%vf*}
+        phys_port_name_pf_index=${phys_port_name_pf_index#pf}
+        if [[ "$phys_port_name_pf_index" != "${pf_port_names[$pf_name]:1}"  ]]; then
+            continue
+        fi
+        echo "Switching VF representor $interface of PF $pf_name to netns $worker_netns"
+        switch_vf $interface $worker_netns
+    done
+}
+
+switch_interface_vfs(){
+    local pf_name="$1"
+    local worker_netns="${2:-$netns}"
+    local pci="$3"
+
+    vfs_list=$(ls /sys/bus/pci/devices/$pci | grep virtfn)
+
+    if [[ -z "${vfs_list}" ]];then
+        echo "Warning: No VFs found for interface $pf_name!!"
+        return 0
+    fi
+
+    for vf in $vfs_list;do
+        local vf_interface="$(ls /sys/bus/pci/devices/$pci/$vf/net)"
+
+        if [[ -n "$vf_interface" ]];then
+            echo "Switching $vf_interface to namespace $worker_netns..."
+            sleep 2
+            if ! switch_vf "$vf_interface" "$worker_netns";then
+                echo "Error: could not switch $vf_interface to namespace $worker_netns!"
+            else
+                echo "Successfully switched $vf_interface to namespace $worker_netns"
+            fi
+        fi
+    done
+}
+
+read_confs(){
+    local conf_file="$1"
+
+    let number_of_netns=$(yq r $conf_file -l)-1
+
+    for index in $(seq 0 $number_of_netns);do
+        netnses+=("$(yq r $conf_file [$index].netns)")
+        let number_of_pfs=$(yq r $conf_file [$index].pfs -l)-1
+        for pf_index in $(seq 0 $number_of_pfs);do
+            pfs[${netnses[-1]}]+="$(yq r $conf_file [$index].pfs[$pf_index]) "
+        done
+    done
+}
+
+variables_check(){
+    local status=0
+
+    check_empty_var "netns"
+    let status=$status+$?
+    check_empty_var "pfs"
+    let status=$status+$?
+
+    return $status
+}
+
+check_empty_var(){
+    local var_name="$1"
+
+    if [[ -z "${!var_name[@]}" ]];then
+        echo "$var_name is empty..."
+        return 1
+    fi
+
+    return 0
+}
+
+main(){
+    local status=0
+
+    while true;do
+        for netns in ${netnses[@]};do
+            switch_pfs "$netns" "${pfs[$netns]}"
+            sleep 2
+            switch_netns_vfs "$netns"
+            sleep 2
+            switch_netns_vf_representors "$netns" "${pfs[$netns]}"
+        done
+        sleep $TIMEOUT
+    done
+    return $status
+}
+
```

Comment:
can you add some logs on the different stages performed here to provide some high level info for the user ?

Reply:
added them to the corresponding functions.